contracts/src/PermissionedStakeTable.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { BN254 } from "bn254/BN254.sol";
+import { EdOnBN254 } from "./libraries/EdOnBn254.sol";
+
+/**
+ * @title SimpleStakeTable
+ * @dev An stake table mapping with owner-only access control.
+ */
+contract PermissionedStakeTable is Ownable {
+    event Added(NodeInfo[]);
+    event Removed(NodeInfo[]);
+
+    error StakerAlreadyExists(BN254.G2Point);
+    error StakerNotFound(BN254.G2Point);
+
+    struct NodeInfo {
+        /// The consensus signing key
+        BN254.G2Point blsVK;
+        /// The consensus signing key. Only used for storage in this contract.
+        EdOnBN254.EdOnBN254Point schnorrVK;
+        /// Is the Node DA Node? Only used for storage in this contract.
+        bool isDA;
+    }
+
+    // State mapping from staker IDs to their staking status
+    mapping(bytes32 nodeID => bool isStaker) private stakers;
+
+    constructor(address initialOwner, NodeInfo[] memory initialStakers) Ownable(initialOwner) {
+        insert(initialStakers);
+    }
+
+    function insert(NodeInfo[] memory newStakers) public onlyOwner {
+        // TODO: revert if array empty
+        for (uint256 i = 0; i < newStakers.length; i++) {
+            bytes32 stakerID = _hashBlsKey(newStakers[i].blsVK);
```

Comment:
should there be a check to validate this blsVK? E.g. ensuring it's non zero. Or perhaps there are some crypto libraries that can be used for validity checks.

Reply:
i'm guessing this is not needed because it's trusted? @sveitser @tbro

Reply:
https://github.com/EspressoSystems/espresso-sequencer/issues/2384